src/Features/Core/Portable/EditAndContinue/DebuggingSession.cs

```diff
@@ -237,48 +238,114 @@ public async Task<(Guid Mvid, Diagnostic? Error)> GetProjectModuleIdAsync(Projec
 
         /// <summary>
         /// Get <see cref="EmitBaseline"/> for given project.
-        /// Must be called on MTA thread.
         /// </summary>
-        /// <returns>Null if the module corresponding to he project hasn't been loaded yet</returns>
-        /// <exception cref="IOException">Error reading project's binary.</exception>
-        public EmitBaseline? GetOrCreateEmitBaseline(ProjectId projectId, Guid mvid, IDebuggeeModuleMetadataProvider debugeeModuleMetadataProvider)
+        /// <returns>True unless the project outputs can't be read.</returns>
+        public bool TryGetOrCreateEmitBaseline(
+            Project project,
+            ArrayBuilder<IDisposable> readers,
+            out ImmutableArray<Diagnostic> diagnostics,
+            [NotNullWhen(true)] out EmitBaseline? baseline)
         {
-            Debug.Assert(Thread.CurrentThread.GetApartmentState() == ApartmentState.MTA, "SymReader requires MTA");
-
-            EmitBaseline? baseline;
             lock (_projectEmitBaselinesGuard)
             {
-                if (_projectEmitBaselines.TryGetValue(projectId, out baseline))
+                if (_projectEmitBaselines.TryGetValue(project.Id, out baseline))
                 {
-                    return baseline;
+                    diagnostics = ImmutableArray<Diagnostic>.Empty;
+                    return true;
                 }
             }
 
-            var moduleInfo = debugeeModuleMetadataProvider.TryGetBaselineModuleInfo(mvid);
```

Comment:
>TryGetBaselineModuleInfo [](start = 59, length = 24)

This API is not available OOP as it passes us a pointer to memory with the module metadata image. We now always read the metadata image from file.